pages/doc-components/CodeContainer.js

```diff
@@ -0,0 +1,63 @@
+import React from 'react'
+import {LiveCode as MDXLive} from 'mdx-live'
+import copy from 'clipboard-copy'
+
+import {Relative, Absolute, Link, StyledOcticon} from '../..'
+import {Clippy, Check} from '@githubprimer/octicons-react'
+
+function CodeContainer({children: code, className: language}) {
+  if (language === 'language-.jsx') return <LiveCode initialCode={code} />
+  else return <StaticCode code={code} />
+}
+
+function LiveCode({initialCode}) {
+  const [code, setCode] = React.useState(initialCode)
+
+  return (
+    <Relative>
+      <MDXLive code={code} editorProps={{onChange: code => setCode(code)}} />
+      <CopyButton code={code} />
+    </Relative>
+  )
+}
+
+function StaticCode({code}) {
+  return (
+    <Relative>
+      <pre>{code}</pre>
+      <CopyButton code={code} />
+    </Relative>
+  )
+}
+
+/* Totally magic number based on what feels-right™ */
+const COPY_RESET_TIME = 1500
+
+function CopyButton({code}) {
+  const [copied, setCopied] = React.useState(false)
+
+  function onClick() {
+    copy(code)
+    setCopied(true)
+  }
+
+  /* Reset copied icon after some time */
+  React.useEffect(
+    function() {
+      setTimeout(function() {
+        if (copied) setCopied(false)
+      }, COPY_RESET_TIME)
```

Comment:
Should this function return a cleanup function that clears this timeout? I think that will prevent this effect from setting the state of an unmounted component.

Reply:
It should!

Would be nicer to use [npmjs.com/use-timeout](npmjs.com/use-timeout) to abstract the boilerplate?

Reply:
you just helped my hacker, what the hell is happening here, are you guys into helping terrorrixe and rob crippled widowers or what the fuck is happening. You just helped a guy who said thanks coloe with a ice cream cone that is my id thief, are you helping him rob me? Terrorize me with pictures of my dead wife and child, because that's what he';s doing, not to mention cfleaning out PayPay a few times, and stealing from my bank. He has my ssn, dln, my7 mothers maiden name for fucks sake, I tried to tell you earlier and thaught you were going to help me, what the fuck????"?"?? catch this prick, (THE REAL CAREY GALLAGHER - CAPTAINTRIPS1969@OUTLOOK.COM)----STOP THIS BASTARD or he will burn some puppies and laugh...what do I have to do??? He could catch me any second and I have noit a clue what this is all about.  HELLP[

Reply:
PLEASE READ

Reply:
PLEASE READ

Reply:
PLEASE READ ABOVE MESSAGE

Reply:
THIS PICTURE IS ME